projects/vdk-core/src/vdk/internal/builtin_plugins/run/data_job.py

```diff
@@ -76,6 +77,9 @@ def run_step(context: JobContext, step: Step) -> StepResult:
                 if step_executed
                 else ExecutionStatus.NOT_RUNNABLE
             )
+        except CancelJobExecutionException as e:
+            # We need to re-raise this exception so it can be handled in the run_job level
+            raise e
```

Comment:
I am not sure about this isn't it better to mark the StepResult with status "canceled" . 

or rather "cancel_requested" as this is the accurate state at this point. 

Then in run_job it will actually cancel everything beloynd.

Otherwise other hook implementation will think it has failed or they'd need to check explicitly for this exception.